### PR TITLE
Fe 01 display all recipes and add routing

### PR DESCRIPTION
**Pull request by jakzda**
Added:
Bootstrap 5.1
Service with http get to get list of recipes and specific recipe
recipe-list component to display list of recipes with paths to component recipe-view
routing module

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -31,7 +31,9 @@
               "src/assets"
             ],
             "styles": [
+              "node_modules/bootstrap/dist/css/bootstrap.min.css",
               "src/styles.scss"
+
```
Comment (Isterius) [inline angular.json:36]: unnecessary empty line

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { RecipeListComponent } from './recipes/recipe-list/recipe-list.component';
+
+import { NgModule } from "@angular/core"
+import { Routes, RouterModule } from "@angular/router"
+
+const appRoutes: Routes = [
+ {path:'', redirectTo: '/recipe-list', pathMatch: 'full'},
+ {path: 'recipe-list', component: RecipeListComponent},
+ {path: 'recipe-view/:id', component: RecipeViewComponent}
+
```
Comment (Isterius) [inline src/app/app-routing.module.ts:11]: unnecessary empty line

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { Routes, RouterModule } from "@angular/router"
+
+const appRoutes: Routes = [
+ {path:'', redirectTo: '/recipe-list', pathMatch: 'full'},
+ {path: 'recipe-list', component: RecipeListComponent},
+ {path: 'recipe-view/:id', component: RecipeViewComponent}
+ 
+]
+
+
```
Comment (Isterius) [inline src/app/app-routing.module.ts:14]: unnecessary empty line

File: src/app/app-routing.module.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+@NgModule({
+    imports: [RouterModule.forRoot(appRoutes)],
+    exports: [RouterModule]
+})
+export class AppRoutingModule{
+
```
Comment (Isterius) [inline src/app/app-routing.module.ts:20]: unnecessary empty line

File: src/app/app.component.ts [hunk 5]
```diff
@@ -6,5 +6,5 @@ import { Component } from '@angular/core';
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
-  title = 'team4-fe-angular';
+  title = 'team4-fe-angularsiema';
```
Comment (Isterius) [inline src/app/app.component.ts:9]: is it used?

**Comment by Isterius** [comment]
@jakzda you probably created this branch from test branch we've created to learn how to create pull-requests :)
Look at commit list, there's our test commit

Let's try to get rid of it together on today's academy 🤔

File: src/app/header/header.component.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<nav class="navbar navbar-expand-lg navbar-dark bg-primary" >
+    <div class="container-fluid">
+      <a class="navbar-brand" routerLink="/recipe-list">
+        <img src="https://i.ibb.co/D7fLR0v/pascal2.jpg" alt="" width="30" height="30" class="d-inline-block align-text-top"> 
+        Pascal Brodnicki Fan Club
+        </a>
```
Comment (Isterius) [inline src/app/header/header.component.html:6]: wrong formatting

Comment (Isterius) [reply src/app/header/header.component.html:6]: opening and closing of one tag should be indented the same

File: src/app/header/header.component.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss']
+})
+export class HeaderComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (Isterius) [inline src/app/header/header.component.ts:13]: no need to keep empty methods (`ngOnInit` & `constructor`)

File: src/app/models/recipe-response.model.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+export interface RecipeResponse{
+    data: Array<SingleRecipeResponse>;
+}
+
+
+
```
Comment (Isterius) [inline src/app/models/recipe-response.model.ts:6]: too many empty lines

File: src/app/models/recipe.model.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+export interface Recipe {
+     id: string;
+     name: string;
+     content: string;
+
+    // constructor(id: number, name: string, content: string){
+    // this.id = id;
+    // this.name = name;
+    // this.content = content;
+    // }
```
Comment (Isterius) [inline src/app/models/recipe.model.ts:10]: not needed probably

File: src/app/recipes/recipe-list/recipe-list.component.html [hunk 10]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (Isterius) [inline src/app/recipes/recipe-list/recipe-list.component.html:1]: not needed empty line

File: src/app/recipes/recipe-list/recipe-list.component.html [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+<div >
+  <h1 style="text-align: center;">All recipes</h1>
```
Comment (Isterius) [inline src/app/recipes/recipe-list/recipe-list.component.html:3]: pls do not use inline styling

File: src/app/recipes/recipe-list/recipe-list.component.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+  templateUrl: './recipe-list.component.html',
+  styleUrls: ['./recipe-list.component.scss']
+})
+export class RecipeListComponent implements OnInit {
+  recipes: Recipe[] = [];
+
+  constructor(private recipeService: RecipeService) { 
+
+    
+
```
Comment (Isterius) [inline src/app/recipes/recipe-list/recipe-list.component.ts:16]: not needed empty lines

File: src/app/recipes/recipe-view/recipe-view.component.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+@Component({
+  selector: 'app-recipe-view',
+  templateUrl: './recipe-view.component.html',
+  styleUrls: ['./recipe-view.component.scss']
+})
+export class RecipeViewComponent implements OnInit {
+  
+  constructor(private route: ActivatedRoute, private recipeService: RecipeService) { }
+
+  recipe: {id:string, name: string, content: string} ={id:'0',name:'0',content:'0'};
```
Comment (Isterius) [inline src/app/recipes/recipe-view/recipe-view.component.ts:15]: pls put more attention to code formatting

also - this approach is good enough for now, but let's discuss other possibilities on Academy :)

File: src/app/recipes/recipe-view/recipe-view.component.ts [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    this.route.params.subscribe((params: Params) => {
+      this.recipe.id = params['id'];
+    })
+
+    this.recipeService.getRecipeById(this.recipe.id).subscribe(
+      (recipe: Recipe)  =>{
+        this.recipe = recipe;
+      })
+      
+
```
Comment (Isterius) [inline src/app/recipes/recipe-view/recipe-view.component.ts:28]: too many empty lines

File: src/app/recipes/recipe.service.ts [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+import {map} from "rxjs/operators";
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RecipeService {
+  private http: HttpClient;
+
+
```
Comment (Isterius) [inline src/app/recipes/recipe.service.ts:15]: not needed empty lines

File: src/app/recipes/recipe.service.ts [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+     .pipe(map(recipe => {
+       return {
+        id: recipe.id,
+        name: recipe.name,
+        content: recipe.content
+       };
+     }))
+
+   }
+
```
Comment (Isterius) [inline src/app/recipes/recipe.service.ts:46]: empty lines :D

**Review by Isterius — CHANGES_REQUESTED** [review]
most of the comments are about code formatting and unnecessary empty lines

File: src/app/models/recipe-response.model.ts [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+export interface SingleRecipeResponse {
+        id: string;
+        name: string;
+        content: string;
+}
```
Comment (Isterius) [inline src/app/models/recipe-response.model.ts:11]: is that full response?

File: src/app/recipes/recipe-view/recipe-view.component.ts [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+}
+}
```
Comment (PatrykTyszka) [inline src/app/recipes/recipe-view/recipe-view.component.ts:30]: and one exception :D you should add empty line at the end of file :D

**Review by Isterius — APPROVED** [review]

**Pull request merged**